ompi/mca/common/ompio/common_ompio_file_open.c

```diff
@@ -100,6 +103,22 @@ int mca_common_ompio_file_open (ompi_communicator_t *comm,
     ompio_fh->f_get_mca_parameter_value=get_mca_parameter_value_fn;
 
     ompio_fh->f_filename = filename;
+    if (opal_path_is_absolute(filename) ) {
+        ompio_fh->f_fullfilename = strdup(filename);
+    }
+    else {
+        char path[256];
```

Comment:
I think this should be `OPAL_PATH_MAX` instead of an arbitrary constant?  I think we're still ok leaving that on the stack, although it's getting a little bit large for some of the lightweight threading systems?

Reply:
incorporate the change